src/services/safes.rs

```diff
@@ -1,11 +1,119 @@
+
+    Ok(safe_state)
+}
+
+fn get_last_collectible(context: &Context, safe_address: &String) -> ApiResult<i64> {
+    let url = format!(
+        "{}/v1/safes/{}/transfers/?\
+        &ordering=executionDate\
+        &limit=1",
+        base_transaction_service_url(),
+        safe_address,
+    );
+
+    let body = RequestCached::new(url)
+        .request_timeout(transaction_request_timeout())
+        .execute(context.client(), context.cache())?;
+    let transaction: Page<Transfer> = serde_json::from_str(&body)?;
+
+    log::debug!("{:#?}", &transaction);
+    transaction
+        .results
+        .get(0)
+        .as_ref()
+        .map(|transfer| match transfer {
+            Transfer::Erc721(transfer) => transfer.execution_date.timestamp(),
+            Transfer::Erc20(transfer) => transfer.execution_date.timestamp(),
+            Transfer::Ether(transfer) => transfer.execution_date.timestamp(),
+            Transfer::Unknown => 0,
+        })
+        .ok_or(api_error!("Couldn't get tx timestamps"))
+}
+
+fn get_last_queued_tx(context: &Context, safe_address: &String) -> ApiResult<i64> {
+    let url = format!(
+        "{}/v1/safes/{}/multisig-transactions/?\
+        &ordering=submissionDate\
+        &executed=false\
+        &trusted=true\
+        &limit=1",
+        base_transaction_service_url(),
+        safe_address,
+    );
+
+    let body = RequestCached::new(url)
+        .request_timeout(transaction_request_timeout())
+        .execute(context.client(), context.cache())?;
+    let transaction: Page<MultisigTransaction> = serde_json::from_str(&body)?;
+
+    log::debug!("{:#?}", &transaction);
+    transaction
+        .results
+        .get(0)
+        .as_ref()
+        .map(|tx| tx.modified.as_ref().map(|it| it.timestamp()))
+        .flatten()
+        .ok_or(api_error!("Couldn't get tx timestamps"))
+}
+
+fn get_last_history_tx(context: &Context, safe_address: &String) -> ApiResult<i64> {
+    let url = format!(
+        "{}/v1/safes/{}/all-transactions/?\
+        &ordering=executionDate
+        &queued=false\
+        &executed=true",
+        base_transaction_service_url(),
+        safe_address
+    );
+
+    let body = RequestCached::new(url)
+        .request_timeout(transaction_request_timeout())
+        .execute(context.client(), context.cache())?;
+    let transaction: Page<Transaction> = serde_json::from_str(&body)?;
+
+    log::debug!("{:#?}", &transaction);
+    transaction
+        .results
+        .get(0)
+        .as_ref()
+        .map(|tx| match tx {
+            Transaction::Multisig(tx) => tx
+                .modified
+                .as_ref()
+                .map(|it| it.timestamp())
+                .unwrap_or(tx.submission_date.timestamp()),
+            Transaction::Ethereum(tx) => tx.execution_date.timestamp(),
+            Transaction::Module(tx) => tx.execution_date.timestamp(),
+            Transaction::Unknown => 0,
+        })
+        .ok_or(api_error!("Couldn't get tx timestamps"))
```

Comment:
Currently we would error for an empty tx list, right? That is probably not optimal :D

Reply:
the caller does a  `unwrap_or(0)`